File: application/src/main/java/com/example/server/controller/ItemController.java
package com.example.server.controller;

import com.example.api.ItemApi;
import com.example.api.dto.ItemDto;
import com.example.api.dto.ItemDtoWithBookings;
import com.example.server.service.ItemService;
import com.example.server.mapper.ItemMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(path = ItemApi.PATH)
@RequiredArgsConstructor
public class ItemController implements ItemApi {

    private final ItemMapper itemMapper;
    private final ItemService itemService;

    @Override
    public ItemDto create(ItemDto itemDto, Long userId) {
        return itemMapper.toItemDto(itemService.create(itemDto, userId));
    }

    @Override
    public ItemDto update(Long id, ItemDto itemDto, Long userId) {
        return itemMapper.toItemDto(itemService.update(id, itemDto, userId));
    }

    @Override
    public ItemDtoWithBookings findById(Long id, Long userId) {
        return itemService.findByIdWithBooking(id, userId);
    }

    @Override
    public List<ItemDtoWithBookings> findAllByOwnerId(Long userId, Integer from, Integer size) {
        return itemService.findAllByOwnerId(userId, from, size);
    }

    @Override
    public List<ItemDto> search(String text, Integer from, Integer size) {
        return itemMapper.toItemDto(itemService.search(text, from, size));
    }
}

File: application/src/main/kotlin/com/example/server/controller/ItemController.kt
package com.example.server.controller

import com.example.api.ItemApi
import com.example.api.model.ItemDto
import com.example.api.model.ItemDtoWithBooking
import com.example.server.mapper.ItemMapper
import com.example.server.service.ItemService
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [ItemApi.PATH])
class ItemController(
    private val itemMapper: ItemMapper,
    private val itemService: ItemService
) : ItemApi {

    override fun create(itemDto: ItemDto, userId: Long): ItemDto {
        return itemMapper.toItemDto(itemService.create(itemDto, userId))
    }

    override fun update(id: Long, itemDto: ItemDto, userId: Long): ItemDto {
        return itemMapper.toItemDto(itemService.update(id, itemDto, userId))
    }

    override fun findById(id: Long, userId: Long): ItemDtoWithBooking {
        return itemService.findByIdWithBooking(id, userId)
    }

    override fun findAllByOwnerId(userId: Long, from: Int, size: Int): List<ItemDtoWithBooking> {
        return itemService.findAllByOwnerId(userId, from, size)
    }

    override fun search(text: String, from: Int, size: Int): List<ItemDto> {
        return itemMapper.toItemDto(itemService.search(text, from, size))
    }
}
